src/worker/keyman-api-service.ts

```diff
@@ -0,0 +1,42 @@
+import { SearchLanguage } from "./models";
+import { KeyboardData } from "./storage-models";
+
+/**
+ * @file KeymanAPI class to implement all services provided by official Keyman API
+ * Note: Currently this class is used for the functionality to fetch all existing keyboard languages
+ * and more function(s) could be added in the future
+ */
+export class KeymanAPI {
+  /**
+   * API configs
+   */
+  baseUrl = "https://api.keyman.com/search";
+  /**
+   * Query all languages data, doc: https://help.keyman.com/developer/cloud/search/1.0/
+   */
+  params = { q: "l" };
+  url: URL;
+  langageArray: Array<KeyboardData>;
```

Comment:
Typo in the name, plus a better name excludes its type (i.e., an array called "Array"):

```suggestion
  languages: KeyboardData[];
```